openassessment/xblock/leaderboard_mixin.py

```diff
@@ -76,16 +76,18 @@ def render_leaderboard_complete(self, student_item_dict):
         for score in scores:
             score['files'] = []
             if 'file_keys' in score['content']:
-                for key in score['content']['file_keys']:
-                    url = ''
-                    try:
-                        url = file_upload_api.get_download_url(key)
-                    except FileUploadError:
-                        pass
-                    if url:
-                        score['files'].append(url)
+                file_keys = score['content'].get('file_keys', [])
+                descriptions = score['content'].get('files_descriptions', [])
+                for idx, key in enumerate(file_keys):
+                    file_download_url = self._get_file_download_url(key)
+                    if file_download_url:
+                        file_description = descriptions[idx] if idx < len(descriptions) else ''
+                        score['files'].append((file_download_url, file_description))
+
             elif 'file_key' in score['content']:
-                score['files'].append(file_upload_api.get_download_url(score['content']['file_key']))
+                file_download_url = self._get_file_download_url(score['content']['file_key'])
+                if file_download_url:
+                    score['files'].append((file_download_url, ''))
```

Comment:
@noraiz-anwar Before multiple file feature was added, we used to treat/add single files. Single uploads don't have any description as well. When the multi-upload feature was added, this changed the way we use to handle files. 

This method is tackling both keys, `file_key` and `file_keys` for the backward compatibility.